Curriculum vitae of Mitchell Jon Stanton-Cook
==============================================


Hackatons etc.
--------------

`Mozilla Science Lab Global Sprint 2015`_ (2015): **UQ Site**. Worked on establishing infrustructure for Australian based library carpentry workshops.

Unearthed_ (2015): **Team RoockMe**. Developed an OpenCV based particle size classifier. Produced a working tool that was live demoed to the judges.

`TANDA hackathon`_ (2015): **Team Dark Magic**. Performed exploratory data analysis of TANDA's time and attendance data using python. Developed an interactive geolocated salary visualization tool. Backend was RethinkDB. Frontend was custom D3.js visualization.

`Startup Health Weekend`_ (2015): **Team Triosk**. Designed, validated and delivered a Kiosk for Emergency Department triage. The Triosk design rationale document is available_. Won_ Best Pitch and Best Collaboronaut.

`Startup Weekend`_: (2015): **Team RentUrWay**. Designed, validated and mocked a privacy aware tenant-landlord communication tool. Simple web development and D3.js charts (for mocks-ups).


.. _Unearthed: http://unearthed.solutions
.. _`TANDA hackathon`: https://www.tanda.co/tanda-open-data-hackathon-this-weekend-17th-18th-april/
.. _`Startup Health Weekend`: http://www.up.co/communities/australia/startup-weekend/4813
.. _`Startup Weekend`: http://www.rivercitylabs.net/event/startup-weekend-brisbane/
.. _`Mozilla Science Lab Global Sprint 2015`: https://www.mozillascience.org/global-sprint-2015
.. _available: http://triosk.co/triosk_overview.pdf
.. _Won: http://www.ilabaccelerator.com/2015/04/its-a-wrap-australias-first-startup-weekend-for-health/



Miscellaneous
-------------

I designed, organised and facilitated the inaugural Australian MIcrobial GenOmics Symposium (AMIGOS_) hackathon. AMIGOS_ was a two day hackathon bringing together 45 Australian based Microbial Genomics researchers.

I am a certified `Software Carpentry Instructor`_. Through Software Carpentry I teach researchers basic software skills(BASH shell, python programming, software revision control & and simple SQL databases). I am the lead instructor for a two day bootcamp `in July`.

I am a co-organiser the `2015 HealthHack`_ in Brisbane. HealthHack is a data hack for medical researchers. The goal (in my opinion) is to bring to focus the importance of software engineers in medical research fields.

I am a member of `River City Labs`_. `River City Labs`_ is a coworking community where members come together to promote and develop early stage and start up businesses and entrepreneurial activity.


.. _AMIGOS: http://theamigos.space
.. _`Software Carpentry Instructor`: http://software-carpentry.org/pages/team.html
.. _`in July`: http://bio-swc-bne.github.io/2015-07-02-UQ/
.. _`2015 HealthHack`: http://www.healthhack.com.au
.. _`River City Labs`: http://www.rivercitylabs.net



Publications
------------
For the most up-to-date list (and metrics) see my `Google Scholar profile`_.


**Molecular characterization of a multidrug resistance IncF plasmid from the globally disseminated Escherichia coli ST131 clone**.
Phan MD, Forde BM, Peters KM, Sarkar S, Hancock S, **Stanton-Cook M**, Ben Zakour NL, Upton M, Beatson SA, Schembri MA.
PLoS One
doi: `10.1371/journal.pone.0122369`_

**Draft Genome Sequence of Pseudomonas fluorescens SRM1, an Isolate from Spoiled Raw Milk**.
Lo R, **Stanton-Cook MJ**, Beatson SA, Turner MS, Bansal N.
Genome Announc
doi: `10.1128/genomeA.00138-15`_

**Third-generation cephalosporin resistance conferred by a chromosomally encoded blaCMY-23 gene in the Escherichia coli ST131 reference strain EC958**.
Phan MD, Peters KM, Sarkar S, Forde BM, Lo AW, **Stanton-Cook M**, Roberts LW, Upton M, Beatson SA, Schembri MA.
J Antimicrob Chemother
doi: `10.1093/jac/dkv066`_

**Molecular analysis of asymptomatic bacteriuria Escherichia coli strain VR50 reveals adaptation to the urinary tract by gene acquisition**.
Beatson SA, Ben Zakour NL, Totsika M, Forde BM, Watts RE, Mabbett AN, Szubert JM, Sarkar S, Phan MD, Peters KM, Petty NK, Alikhan NF, Sullivan MJ, Gawthorne JA, **Stanton-Cook M**, Nhu NT, Chong TM, Yin WF, Chan KG, Hancock V, Ussery DW, Ulett GC, Schembri MA.
Infect Immun
doi: `10.1128/IAI.02810-14`_

**The complete genome sequence of Escherichia coli EC958: a high quality reference sequence for the globally disseminated multidrug resistant E. coli O25b:H4-ST131 clone**.
Forde BM, Ben Zakour NL, **Stanton-Cook M**, Phan MD, Totsika M, Peters KM, Chan KG, Schembri MA, Upton M, Beatson SA.
PLoS One
doi: `10.1371/journal.pone.0104400`_

**Global dissemination of a multidrug resistant Escherichia coli clone**.
Petty NK, Ben Zakour NL, **Stanton-Cook M**, Skippington E, Totsika M, Forde BM, Phan MD, Gomes Moriel D, Peters KM, Davies M, Rogers BA, Dougan G, Rodriguez-Ba√±o J, Pascual A, Pitout JD, Upton M, Paterson DL, Walsh TR, Schembri MA, Beatson SA.
Proc Natl Acad Sci USA
doi: `10.1073/pnas.1322678111`_

**uPEPperoni: an online tool for upstream open reading frame location and analysis of transcript conservation**.
Skarshewski A, **Stanton-Cook M**, Huber T, Al Mansoori S, Smith R, Beatson SA, Rothnagel JA.
BMC Bioinformatics
doi: `10.1186/1471-2105-15-36`_

**Engineering [Ln(DPA)3] 3- binding sites in proteins: a widely applicable method for tagging proteins with lanthanide ions**.
Jia X, Yagi H, Su XC, **Stanton-Cook M**, Huber T, Otting G.
J Biomol NMR
doi: `10.1007/s10858-011-9529-x`_

**Generation of pseudocontact shifts in protein NMR spectra with a genetically encoded cobalt(II)-binding amino acid**.
Nguyen TH, Ozawa K, **Stanton-Cook M**, Barrow R, Huber T, Otting G.
Angew Chem Int Ed Engl
doi: `10.1002/anie.201005672`_

**Tunable paramagnetic relaxation enhancements by [Gd(DPA)(3)] (3-) for protein structure analysis**.
Yagi H, Loscha KV, Su XC, **Stanton-Cook M**, Huber T, Otting G.
J Biomol NMR
doi: `10.1007/s10858-010-9416-x`_

**Numbat: an interactive software tool for fitting Deltachi-tensors to molecular coordinates using pseudocontact shifts**.
Schmitz C, **Stanton-Cook MJ**, Su XC, Otting G, Huber T.
J Biomol NMR
doi: `10.1007/s10858-008-9249-z`_


.. _`Google Scholar profile`: https://scholar.google.com.au/citations?user=MGafrX4AAAAJ&hl=en
.. _`10.1371/journal.pone.0122369`: http://doi.org/10.1371/journal.pone.0122369
.. _`10.1128/genomeA.00138-15`: http://doi.org/10.1128/genomeA.00138-15
.. _`10.1093/jac/dkv066`: http://doi.org/10.1093/jac/dkv066
.. _`10.1128/IAI.02810-14`: http://doi.org/10.1128/IAI.02810-14
.. _`10.1371/journal.pone.0104400`: http://doi.org/10.1371/journal.pone.0104400
.. _`10.1073/pnas.1322678111`: http://doi.org/10.1073/pnas.1322678111
.. _`10.1186/1471-2105-15-36`: http://doi.org/10.1186/1471-2105-15-36
.. _`10.1007/s10858-011-9529-x`: http://doi.org/10.1007/s10858-011-9529-x
.. _`10.1002/anie.201005672`: http://doi.org/10.1002/anie.201005672
.. _`10.1007/s10858-010-9416-x`: http://doi.org/10.1007/s10858-010-9416-x
.. _`10.1007/s10858-008-9249-z`: http://doi.org/10.1007/s10858-008-9249-z



References
----------

Provided on request.
